R/run_grid.R

```diff
@@ -189,6 +201,34 @@ compare_baseline_xpvwrh <- function(y, baseline, uptake_first_dose,
   names(ret) <- paste0("inc_", names(flows))
   ret <- c(flows, ret)
 
+  ## extract number under vaccine protection
+      vacsnap <- list()
+     # fully vaccine protected, snapshot of N in: V(3) and R(5)
+        vacsnap$vacprotec_full <-
+        t(aggregate(y, "N", stratum = c(3, 5)))
+
+    # partially vaccine protected, snapshot of N in: P(2)
+        vacsnap$vacprotec_part <-
+         t(aggregate(y, "N", stratum = 2))
+
+    # vaccine protected total, snapshot of N in: P(2), V(3), W(4)
+        vacsnap$vacprotec_total <-
+        t(aggregate(y, "N", stratum = c(2, 3, 5)))
+
+    # remove the 0
+        vacsnap <- lapply(vacsnap, "[", -1, )
+
+  ret <- c(vacsnap, ret)
+
+  ## calculate proportion of the population under vaccine protection
+    # get total pop size  (including H!) This should be the same across
+    # all model runs for all timepoints
+    N <- t(aggregate(y, "N"))[1, 1]
```

Comment:
```suggestion
    N <- aggregate(y, "N")[, -1]
```
This should work fine with the object dimensions as N will be a vector of length t, and `ret$vacprotec_full` etc will have dim `c(t, n_par)` (I think) and it means the code will still work if we ever moved to having a changing population size